sky/spot/controller.py

```diff
@@ -88,12 +98,43 @@ def _run(self):
                 3. Any unexpected error happens during the `sky.launch`.
         Other exceptions may be raised depending on the backend.
         """
-        logger.info(f'Started monitoring spot job {self._job_id}, '
-                    f'name: {self._task_name!r}.')
-        spot_state.set_starting(self._job_id)
+        if task.run is None:
+            logger.info(f'Sub job {task_id} ({task.name}) is skipped '
+                        'for empty run.')
+            spot_state.set_started(self._job_id,
+                                   task_id,
+                                   start_time=time.time())
+            spot_state.set_succeeded(self._job_id,
+                                     task_id,
+                                     end_time=time.time())
+            return True
+        job_id_env_var = task.envs[constants.JOB_ID_ENV_VAR]
+        submitted_at = time.time()
+        if task_id == 0:
+            submitted_at = self._backend.run_timestamp
+        spot_state.set_submitted(
```

Comment:
How/where do we make sure the "job row" in `spot queue` show the latest task's status? Actually it shouldn't blindly follow latest task task (as task N+1 will go through SUBMITTED, and we don't want to reset job row's status to SUBMITTED).